src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/controller/ZipStatusControllerTest.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.controller;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.hmcts.reform.bulkscanprocessor.controllers.ZipStatusController;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.out.zipfilestatus.ZipFileEnvelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.out.zipfilestatus.ZipFileEvent;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.out.zipfilestatus.ZipFileStatus;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.zipfilestatus.ZipFileStatusService;
+import uk.gov.hmcts.reform.bulkscanprocessor.util.DateFormatter;
+
+import java.sql.Timestamp;
+import java.util.List;
+
+import static java.time.Instant.now;
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(ZipStatusController.class)
+public class ZipStatusControllerTest {
+
+    @MockBean
+    private ZipFileStatusService service;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    public void should_return_data_returned_from_the_service() throws Exception {
+
+        List<ZipFileEnvelope> envelopes = asList(
+            new ZipFileEnvelope("0", "container0", "status0"),
+            new ZipFileEnvelope("1", "container1", "status1")
+        );
+
+        List<ZipFileEvent> events = asList(
+            new ZipFileEvent("type0", "container0", Timestamp.from(now().minusSeconds(10))),
+            new ZipFileEvent("type1", "container1", Timestamp.from(now().minusSeconds(15)))
+        );
+
+        given(service.getStatusFor("hello.zip")).willReturn(new ZipFileStatus(envelopes, events));
+
+        mockMvc
```

Comment:
Can we do the assets with actual values what we expect ? I am not sure if the order of envelopes is maintained ?
```
 mockMvc.perform(get("/post/list-all"))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$.*", hasSize(1)))
                .andExpect(jsonPath("$.blogPosts.*", hasSize(2)))
                .andExpect(jsonPath("$.blogPosts[0].id", equalTo(21)))
                .andExpect(jsonPath("$.blogPosts[0].title", equalTo("Spring Boot Testing Improvements")))
                .andExpect(jsonPath("$.blogPosts[0].content", equalTo("New @SpringBootTest and other new annotations those come within Spring Boot 1.4.0.")))
                .andExpect(jsonPath("$.blogPosts[0].author", equalTo("Richard Roe")))
                .andExpect(jsonPath("$.blogPosts[0].createdDate", equalTo(new DateTime(2016, 12, 14, 0, 0).getMillis())))
                .andExpect(jsonPath("$.blogPosts[1].id", equalTo(57)))
                .andExpect(jsonPath("$.blogPosts[1].title", equalTo("Unit Testing")))
                .andExpect(jsonPath("$.blogPosts[1].content", equalTo("Do not push any production code without unit tests.")))
                .andExpect(jsonPath("$.blogPosts[1].author", equalTo("John Doe")))
                .andExpect(jsonPath("$.blogPosts[1].createdDate", equalTo(new DateTime(2016, 11, 15, 0, 0).getMillis())));
```